### PR TITLE
binary: Improve error-handling behavior (#283)

### PR DESCRIPTION
**Pull request by dvrkps**

### CODE REVIEW

File: exercises/binary/binary_test.go [hunk 1]
```diff
@@ -39,10 +39,14 @@ var testCases = []struct {
 }
 
 func TestParseBinary(t *testing.T) {
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
```
Comment (petertseng) [inline exercises/binary/binary_test.go]: Our mistake for not testing that. I am considering asking for it in a separate commit, but realize that that is a bit paradoxical because it *could* break existing solutions. I don't want to overthink it, though I can't think about it right now. In a few hours.

File: exercises/binary/binary_test.go [hunk 2]
```diff
@@ -17,12 +17,12 @@ import (
...
 // the targetTestVersion here.
 
-const targetTestVersion = 1
+const targetTestVersion = 2
 
 var testCases = []struct {
 	binary   string
 	expected int
-	ok bool
+	ok       bool
```
Comment (petertseng) [inline exercises/binary/binary_test.go]: probably good to put in a separate commit since this change is not to do with what  is wanted in #283 .

**Comment by petertseng** [comment]
I did end up splitting https://github.com/exercism/xgo/pull/361/commits/bd156a4b2958e13212480d9755b5bb50b3abd5be into a separate commit but kept the `testVersion` check.

Thanks!

**Comment by dvrkps** [comment]
Don't forget to close issue.

**Comment by petertseng** [comment]
Ah right.

You could put "Closes #283" into commit message next time to help me with that
